programming_examples/basic/shuffle_transpose/test.cpp

```diff
@@ -0,0 +1,155 @@
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 AMD Inc.
+
+#include <cassert>
+#include <cstring>
+#include <fstream>
+#include <iomanip>
+
+#include "xrt/xrt_bo.h"
+#include "xrt/xrt_device.h"
+#include "xrt/xrt_kernel.h"
+
+#ifndef XCLBIN
+#define XCLBIN "build/final.xclbin"
+#endif
+
+#ifndef INSTS_BIN
+#define INSTS_BIN "build/insts.bin"
+#endif
+
+#ifndef KERNEL_NAME
+#define KERNEL_NAME "MLIR_AIE"
+#endif
+
+/* This example performs a 16x16 INT8 transpose.
+   kernel.cc includes an AIE kernel that is specific to 16x16 */
+
+
+#define IN_SIZE (M * N * sizeof(uint8_t))  // in bytes
+#define OUT_SIZE (M * N * sizeof(uint8_t)) // in bytes
+
+void print_matrix(uint8_t *buf, int n_rows, int n_cols) {
+  for (int row = 0; row < n_rows; row++) {
+    for (int col = 0; col < n_cols; col++) {
+      std::cout << std::setw(4) << int(buf[row * n_cols + col]) << " ";
+    }
+    std::cout << std::endl;
+  }
+}
+
+std::vector<uint32_t> load_instr_binary(std::string instr_path) {
+  // Open file in binary mode
+  std::ifstream instr_file(instr_path, std::ios::binary);
+  if (!instr_file.is_open()) {
+    throw std::runtime_error("Unable to open instruction file\n");
+  }
+
+  // Get the size of the file
+  instr_file.seekg(0, std::ios::end);
+  std::streamsize size = instr_file.tellg();
+  instr_file.seekg(0, std::ios::beg);
+
+  // Check that the file size is a multiple of 4 bytes (size of uint32_t)
+  if (size % 4 != 0) {
+    throw std::runtime_error("File size is not a multiple of 4 bytes\n");
+  }
+
+  // Allocate vector and read the binary data
+  std::vector<uint32_t> instr_v(size / 4);
+  if (!instr_file.read(reinterpret_cast<char *>(instr_v.data()), size)) {
+    throw std::runtime_error("Failed to read instruction file\n");
+  }
+  return instr_v;
+}
```

Comment:
Import this from test_utils: 

https://github.com/Xilinx/mlir-aie/blob/0d7922324a730d8d9088ac9ccfda868d3c419fd5/runtime_lib/test_lib/test_utils.cpp#L96-L111